pilot/pkg/networking/core/v1alpha3/route/retry/retry.go

```diff
@@ -0,0 +1,111 @@
+// Copyright 2018 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package retry
+
+import (
+	"net/http"
+	"strconv"
+	"strings"
+
+	"github.com/envoyproxy/go-control-plane/envoy/api/v2/route"
+	"github.com/gogo/protobuf/types"
+
+	networking "istio.io/api/networking/v1alpha3"
+	"istio.io/istio/pilot/pkg/networking/util"
+)
+
+// DefaultPolicy gets a copy of the default retry policy.
+func DefaultPolicy() *route.RouteAction_RetryPolicy {
+	policy := route.RouteAction_RetryPolicy{
+		NumRetries:           &types.UInt32Value{Value: 10},
+		RetryOn:              "connect-failure,refused-stream,unavailable,cancelled,resource-exhausted",
+		RetriableStatusCodes: []uint32{http.StatusServiceUnavailable},
+		RetryHostPredicate: []*route.RouteAction_RetryPolicy_RetryHostPredicate{
+			{
+				// to configure retries to prefer hosts that haven’t been attempted already,
+				// the builtin `envoy.retry_host_predicates.previous_hosts` predicate can be used.
+				Name: "envoy.retry_host_predicates.previous_hosts",
+			},
+		},
+		HostSelectionRetryMaxAttempts: 3,
```

Comment:
shouldn't this be also 10?

Reply:
I'm not sure we need to change this.  @PiotrSikora any thoughts?

Reply:
Sorry for the late reply, this is fine as it-is. This is a different kind of retries that apply to the host selection in the load balancer and not the request retries at the network layer.